### PR TITLE
chore(storage): upload api and indentation nits

### PR DESCRIPTION
**Pull request by kvramyasri7**
#### Description of changes:

1. uploadData api
2. indentation nits

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/storage/js/copy.mdx [hunk 1]
```diff
@@ -51,19 +51,21 @@ To copy file across different access levels, you'll need to explicitly provide t
 ```typescript
 import { copy, list } from 'aws-amplify/storage';
 
-console.log(await list({ path : 'copied/', options: { accessLevel: 'private' }})); // []
+console.log(
+	await list({ path: 'copied/', options: { accessLevel: 'private' } })
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/copy.mdx:55]:
```suggestion
	await list({ prefix: 'copied/', options: { accessLevel: 'private' } })
```

File: src/fragments/lib-v1/storage/js/list.mdx [hunk 2]
```diff
@@ -85,12 +88,18 @@ This places each item's data inside a special `__data` key.
...
-```javascript
+
+```typescript
 import { list } from 'aws-amplify/storage';
 
-list('photos/', { level: 'protected' })
-  .then(({ results }) => console.log(results))
-  .catch((err) => console.log(err));
+try {
+  const response = list({ prefix: 'photos/', options:  {
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/list.mdx:96]:
```suggestion
  const response = await list({ prefix: 'photos/', options:  {
```

File: src/fragments/lib-v1/storage/js/remove.mdx [hunk 3]
```diff
@@ -5,41 +5,36 @@ Delete stored data from the storage bucket.
...
-	try {
-		await remove({ key: filename, options: { accessLevel: 'protected' } });
-	} catch (error) {
-		console.log('Error ', error);
-	}
-}; 
+import { remove } from 'aws-amplify/storage';
+
+try {
+	await remove({ key: filename, options: { accessLevel: 'protected' } });
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/remove.mdx:24]: I haven't build this branch locally but the indentation here looks different. But I'm totally OK if they look the same in the doc site.

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 4]
```diff
@@ -1,173 +1,194 @@
...
+		},
+	}).result;
+	console.log('Key from Response: ', result.key);
+} catch (error) {
+	console.log('Error : ', error);
+}
 ```
 
-### Monitor progress of upload
+### Protected level of same user
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/upload.mdx:43]:
```suggestion
### Protected level
```

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 5]
```diff
@@ -1,173 +1,194 @@
...
+		},
+	}).result;
+	console.log('Key from Response: ', result.key);
+} catch (error) {
+	console.log('Error : ', error);
+}
 ```
 
-### Encrypted uploads
+### Protected level of another user
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/upload.mdx:62]: This is no-longer supported. Can you remove this section?

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 6]
```diff
@@ -1,173 +1,194 @@
...
+
+try {
+	const result = uploadData({
+    key: filename,
+    data: file,
+    options: {
+      onProgress: ({ transferredBytes, totalBytes }) => {
+        if (totalBytes) {
+          console.log(
+            `Upload progress ${(transferredBytes / totalBytes) * 100} %`
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/upload.mdx:117]:
```suggestion
            `Upload progress ${(Math.round(transferredBytes / totalBytes)) * 100} %`
```

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 7]
```diff
@@ -1,173 +1,194 @@
...
-			console.error('Unexpected error while uploading', err);
-		}
-})
 ```
 
-## Browser uploads
+<!-- TODO Rewrite why do we not have multi-part upload documented for JS? -->
 
-Upload an image in the browser:
+When a page refresh occurs during the upload, re-initializing the upload with the same file will continue from previous progress point.
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/upload.mdx:181]:
```suggestion
When a page refresh occurs during the upload, re-initializing the upload with the same file will continue from previous break point.
```

File: src/fragments/lib-v1/storage/js/download.mdx [hunk 8]
```diff
@@ -71,10 +73,17 @@ Download a file to in-memory buffer.
...
 
 // Download a file from s3 bucket
-const { body, eTag } = await downloadData({ key, data: file, options: {
-onProgress, // Optional progress callback.
-} }).result;
-```
+const { body, eTag } = await downloadData({
+	key,
+	data: file,
+	options: {
```
Comment (stocaaro) [inline src/fragments/lib-v1/storage/js/download.mdx:79]: Up to here this seems like it's an executable call, but from here on it looks like typing?

File: src/fragments/lib-v1/storage/js/get-properties.mdx [hunk 9]
```diff
@@ -1,23 +1,25 @@
...
-		});
-    console.log('File Properties ', result);
-  } catch (error) {
-    console.log('Error ', error);
-  }
+try {
+	const result = await getProperties({
+		key: 'key',
+		options: {
+			accessLevel: 'private' | 'protected' | 'guest', // defaults to `guest`
```
Comment (stocaaro) [inline src/fragments/lib-v1/storage/js/get-properties.mdx:10]: Similar, is this Al executable code or type definition?

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 10]
```diff
@@ -1,173 +1,194 @@
...
+import { uploadData } from 'aws-amplify/storage';
 
-```javascript
-const result = await Storage.put("test.txt", "Hello");
+try {
+	const result = await uploadData({
+		key: filename,
+		data: file,
+		options: {
+			accessLevel: 'private' | 'protected' | 'guest', // defaults to `guest`
```
Comment (stocaaro) [inline src/fragments/lib-v1/storage/js/upload.mdx:13]: Here too

File: src/fragments/lib-v1/storage/js/download.mdx [hunk 11]
```diff
@@ -12,10 +12,11 @@ const getUrlResult: {
 	expiresAt: Date;
 } = await getUrl({
     key: filename,
-    options: { accessLevel: 'private' | 'protected' | 'guest' , // defaults to `guest` 
+    options: { accessLevel?: 'private' | 'protected' | 'guest' , // defaults to `guest`
```
Comment (stocaaro) [inline src/fragments/lib-v1/storage/js/download.mdx:15]: One last case of calls vs types

Comment (kvramyasri7) [reply src/fragments/lib-v1/storage/js/download.mdx]: updated please take a look at it now

**Review by stocaaro — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**